### PR TITLE
Avoid creating foreign keys with the same relational name

### LINKED ISSUE #5769: Duplicate Foreign Key Constraints created in Migrations with TPH inheritance
### Steps to reproduce

    public abstract class Schedule
    {
        public Guid Id { get; set; }
        //...
    }

    public class EmployeeSchedule : Schedule
    {
        public User Owner { get; set; }
        //...
    }

    public class ManagerSchedule : Schedule
    {
        public Manager Owner { get; set; }
        //...
    }

    public abstract class User
    {
        public Guid Id { get; set; }
        //...
    }

    public class Employee : User 
    {
        //...
    }
    public class Manager : User
    {
        //...
    }

### The issue
Within the generated migration:

            migrationBuilder.CreateTable(
                name: "Schedules",
                columns: table => new
                {
                    Id = table.Column<Guid>(nullable: false),
                    OwnerId = table.Column<Guid>(nullable: true),
                    //...
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_Schedules", x => x.Id);
                    table.ForeignKey(
                        name: "FK_Schedules_Users_OwnerId",
                        column: x => x.OwnerId,
                        principalTable: "Users",
                        principalColumn: "Id",
                        onDelete: ReferentialAction.Restrict);
                    table.ForeignKey(
                        name: "FK_Schedules_Users_OwnerId",
                        column: x => x.OwnerId,
                        principalTable: "Users",
                        principalColumn: "Id",
                        onDelete: ReferentialAction.Restrict);
                    //...

### Further technical details

EF Core version: 1.0.0-rc2-final
Operating system: Win 7 x64
Visual Studio version: VS 2015 Update 2

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Validate that matching foreign keys and indexes are compatible

Fixes #5769

### CODE REVIEW

**Comment by AndriySvyryd** [comment]
@rowanmiller please review the exception messages

File: src/Microsoft.EntityFrameworkCore.Relational/Properties/RelationalStrings.resx [hunk 1]
```diff
@@ -277,4 +277,25 @@
...
     <value>{conflictingConfiguration} cannot be set for '{property}', because {existingConfiguration} is already set.</value>
   </data>
+  <data name="DuplicateForeignKeyColumnMismatch" xml:space="preserve">
+    <value>The foreign keys {index1} on '{entityType1}' and {index2} on '{entityType2}' are both mapped to '{table}.{foreignKeyName}' but with different columns ({columnNames1} and {columnNames2}).</value>
+  </data>
+  <data name="DuplicateForeignKeyDeleteBehaviorMismatch" xml:space="preserve">
+    <value>The foreign keys {index1} on '{entityType1}' and {index2} on '{entityType2}' are both mapped to '{table}.{foreignKeyName}' but with different delete behavior ('{deleteBehavior1}' and '{deleteBehavior2}').</value>
+  </data>
+  <data name="DuplicateForeignKeyPrincipalColumnMismatch" xml:space="preserve">
+    <value>The foreign keys {index1} on '{entityType1}' and {index2} on '{entityType2}' are both mapped to '{table}.{foreignKeyName}' but referencing different principal columns ({principalColumnNames1} and {principalColumnNames2}).</value>
```
Comment (rowanmiller) [inline src/Microsoft.EntityFrameworkCore.Relational/Properties/RelationalStrings.resx]: I think maybe "target columns" rather than "principal columns"?

Comment (rowanmiller) [inline src/Microsoft.EntityFrameworkCore.Relational/Properties/RelationalStrings.resx]: Actually, the Fluent API for this concept is `HasPrincipleKey(...)` so I think it's fine as-is.

**Comment by rowanmiller** [comment]
:shipit: from an exception message point of view

**Comment by ajcvickers** [comment]
:shipit:

**Pull request merged**